### PR TITLE
Fix a couple of missing cases of retry on corruption

### PR DESCRIPTION
**Pull request by anand1976**
For SST checksum mismatch corruptions in the read path, RocksDB retries the read if the underlying file system supports verification and reconstruction of data (`FSSupportedOps::kVerifyAndReconstructRead`). There were a couple of places where the retry was missing - reading the SST footer and the properties block. This PR fixes the retry in those cases.

Test plan:
Add new unit tests

### CODE REVIEW

File: table/meta_blocks.cc [hunk 1]
```diff
@@ -262,182 +262,229 @@ Status ReadTablePropertiesHelper(
...
+          PersistentCacheOptions::kEmpty, memory_allocator);
+      s = block_fetcher.ReadBlockContents();
+      if (!s.ok()) {
+        return s;
+      }
+      assert(block_fetcher.GetBlockSizeWithTrailer() == len);
+      TEST_SYNC_POINT_CALLBACK("ReadTablePropertiesHelper:0",
+                               &block_contents.data);
+    } else {
+      // If retrying, use a stronger file system read to check and correct
```
Comment (jaykorean) [inline table/meta_blocks.cc:294]: wondering if `assert(s.IsCorruption());` might be worth adding here.

File: table/format.cc [hunk 2]
```diff
@@ -543,6 +547,29 @@ Status ReadFooterFromFile(const IOOptions& opts, RandomAccessFileReader* file,
...
+    IOOptions new_opts = opts;
+    new_opts.verify_and_reconstruct_read = true;
+    s = ReadFooterFromFileInternal(new_opts, file, fs, prefetch_buffer,
+                                   file_size, footer,
+                                   enforce_table_magic_number);
+    RecordTick(stats, FILE_READ_CORRUPTION_RETRY_COUNT);
+    if (s.ok()) {
+      RecordTick(stats, FILE_READ_CORRUPTION_RETRY_SUCCESS_COUNT);
+    }
+  }
```
Comment (jaykorean) [inline table/format.cc:569]: Looks like we are no longer recording `SST_FOOTER_CORRUPTION_COUNT` tick even when the retry fails. Is that intentional?

Comment (anand1976) [reply table/format.cc:570]: Good catch! We should record it after the retry fails (or after first try if there's no retry).

**Review by jaykorean — APPROVED** [review]
LGTM! 

One nit request would be adding `SST_FOOTER_CORRUPTION_COUNT` check to the unit test that you added in this PR.

**Pull request closed without merging**